Description: "Debian specific" changes: /etc/vpnc/vpnc-script-post-[dis]connect-action
Author: Eduard Bloch <blade@debian.org>
Forwarded: no

--- a/vpnc-script
+++ b/vpnc-script
@@ -557,6 +558,27 @@
 }
 
 do_connect() {
+      # Debian specific, insert your code there to avoid modification of
+      # conffiles like this script
+      if [ -r /etc/vpnc/vpnc-script-connect-action ] ; then
+	 . /etc/vpnc/vpnc-script-connect-action
+      fi
+      ## end Debian specific
+
 	if [ -n "$CISCO_BANNER" ]; then
 		echo "Connect Banner:"
 		echo "$CISCO_BANNER" | while read LINE ; do echo "|" "$LINE" ; done
@@ -607,12 +629,30 @@
 		set_ipv6_default_route
 	fi
 
 	if [ -n "$INTERNAL_IP4_DNS" ]; then
 		$MODIFYRESOLVCONF
+
+	if [ -r /etc/vpnc/vpnc-script-post-connect-action ] ; then
+	   . /etc/vpnc/vpnc-script-post-connect-action
 	fi
+
 }
 
 do_disconnect() {
+
+	# Debian specific, insert your code there to avoid modification of
+	# conffiles like this script
+	if [ -r /etc/vpnc/vpnc-script-disconnect-action ] ; then
+	   . /etc/vpnc/vpnc-script-disconnect-action
+	fi
+
 	if [ -n "$CISCO_SPLIT_INC" ]; then
 		i=0
 		while [ $i -lt $CISCO_SPLIT_INC ] ; do
@@ -655,8 +695,17 @@
 
 	del_vpngateway_route
 
 	if [ -n "$INTERNAL_IP4_DNS" ]; then
 		$RESTORERESOLVCONF
+	if [ -r /etc/vpnc/vpnc-script-post-disconnect-action ] ; then
+	   . /etc/vpnc/vpnc-script-post-disconnect-action
 	fi
 	destroy_tun_device
 }
